### PR TITLE
nyaasi: improve season and episode parsing

### PR DESCRIPTION
**Pull request by ilike2burnthing**
Follow up to https://github.com/Jackett/Jackett/pull/14261 and https://github.com/Prowlarr/Prowlarr/issues/1052.

### CODE REVIEW

File: src/Jackett.Common/Definitions/nyaasi.yml [hunk 1]
```diff
@@ -166,17 +164,33 @@ search:
...
+        - name: re_replace
+          args: ["(?i)\\b(\\d+)(st|nd|rd|th) Season\\b", "$0 S$1"]
+        - name: re_replace
+          args: ["(?i)\\b(S\\d+(?:-\\d+)?) - (\\d+)[\\s\\-~\\+àa&]+(\\d+)\\b", "$1E$2-$3"]
+        - name: re_replace
+          args: ["(?i)\\b(S\\d+(?:-\\d+)?) - (\\d+)\\b", "$1E$2"]
+        - name: re_replace
+          args: ["(?i)\\b(.+? - )(\\d+)[\\s\\-~\\+àa&]+(\\d+)( [\\[\\(])\\b", "$1S01E$2-$3$4"]
+        - name: re_replace
+          args: ["(?i)\\b(.+? - )(\\d+)( [\\[\\(])\\b", "$1S01E$2$3"]
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/nyaasi.yml:189]: Saw a trend of `Title - 01 [1080p]` or `Title - 17-24 (WEBRiP)` so using it identify S01 releases. While it's likely to misidentify, I couldn't actually find any cases. Could use `\\d{1,3}` to make that less likely, or just remove altogether.

**Comment by ilike2burnthing** [comment]
> btw aren't you pushing directly to this repo? 😋

I could, but I prefer to hide my shame over on my own 😋 I think I just got into a habit of doing it that way before I became a maintainer.

File: src/Jackett.Common/Definitions/nyaasi.yml [hunk 2]
```diff
@@ -166,22 +164,42 @@ search:
...
+        - name: re_replace
+          args: ["(?i)\\b(?:S\\s|Seasons?\\s?)(\\d+(?:-\\d+)?)\\b(?!(?:-\\d+)?\\s(?:EP|Episodes?)?\\s?(?:\\d+(?:-\\d+)?)?\\s?S\\d+(?:E\\d+(?:-\\d+)?)?)", "$0 S$1"]
+        - name: re_replace
+          args: ["(?i)\\b(?:EP|Episodes?)\\s?(\\d+(?:-\\d+)?)\\b(?!(?:-\\d+)?\\sS\\d+(?:-\\d+)?(?:E\\d+(?:-\\d+)?)?)", "$0 S01E$1"]
     title_has_season:
       text: "{{ .Result.title_optional }}"
       filters:
         - name: regexp
-          args: "(?i)\\b(S\\d{1,3}(E\\d+)?)\\b"
+          args: "(?i)(S\\d{1,3}(E\\d+)?)"
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/nyaasi.yml:202]: Titles with underscores weren't being picked up by this. Alternatively could do:

```suggestion
          args: ""(?i)\\b.+_?(S\\d{1,3}(E\\d+)?)_?.+\\b""
```

**Comment by TopoR** [comment]
Yep, that was me 😅

My issue only problematic with the **One Piece** for now. All output with additional SXEY format in-between absolute order and series name results Sonarr to mark this release as _Unknown Series_.

**Comment by TopoR** [comment]
> Restart prowlarr and try again.
Thanks! There also a new update today. Checked and it reversed back.

**Pull request merged**